src/warpychain/initial_syncer.go

```diff
@@ -0,0 +1,258 @@
+package warpychain
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/warp-contracts/syncer/src/utils/config"
+	"github.com/warp-contracts/syncer/src/utils/monitoring"
+	"github.com/warp-contracts/syncer/src/utils/task"
+	"gorm.io/gorm"
+)
+
+type InitialSyncer struct {
+	db *gorm.DB
+	*task.Task
+	warpyChainClient *WarpyChainClient
+
+	maxElapsedTime time.Duration
+	maxInterval    time.Duration
+	monitor        monitoring.Monitor
+}
+
+func NewInitialSyncer(config *config.Config) (self *InitialSyncer) {
```

Comment:
You should divide this task and handle initial sync the same way you handle regular sync.

Syncer and contractor handle this case by having a task that produces height that should be processed. The rest of the code doesn't know if it's an init sync or a regular one.